gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/testing/LocalDatastoreHelper.java

```diff
@@ -555,7 +555,7 @@ private static int findAvailablePort() {
 
   /**
    * Returns a {@link DatastoreOptions} instance that sets the host to use the Datastore emulator
-   * on localhost.
+   * on localhost. This should only be called after calling {@link #start}.
```

Comment:
As I mentioned above I think here we should detected that the emulator has not started yet and we should throw an `IllegalStateException` with meaningful message.